Generator generated an invalid version config for maven-enforcer-plugin in pom.xml

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

The generated maven-enforcer-plugin version config seems wrong. Here's an example:
```
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-enforcer-plugin</artifactId>
    <version>${maven-enforcer-plugin.version}</version>
    <executions>
        <execution>
            <id>enforce-versions</id>
            <goals>
                <goal>enforce</goal>
            </goals>
        </execution>
        <execution>
            <id>enforce-dependencyConvergence</id>
            <configuration>
                <rules>
                    <DependencyConvergence />
                </rules>
                <fail>false</fail>
            </configuration>
            <goals>
                <goal>enforce</goal>
            </goals>
        </execution>
    </executions>
    <configuration>
        <rules>
            <requireMavenVersion>
                <message>You are running an older version of Maven. JHipster requires at least Maven ${maven.version}</message>
                <version>[${maven.version},)</version>
            </requireMavenVersion>
            <requireJavaVersion>
                <message>You are running an incompatible version of Java. JHipster supports JDK 17 to 11.</message>
                <version>[17,18),[16,17),[15,16),[14,15),[13,14),[12,13),[11,12)</version>
            </requireJavaVersion>
        </rules>
    </configuration>
</plugin>
```
I think it should be `[11,12),[12,13),[13,14),[14,15),[15,16),[16,17),[17,18)`

##### **Motivation for or Use Case**

Project built failed because of the reversed version config:

```
mvnw package -Pprod verify jib:dockerBuild
[INFO] Scanning for projects...
[INFO]
[INFO] ---------< com.inspur.ihealth.qa.manage:qa-manage-application >---------
[INFO] Building QA Manage Application 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-resources-plugin:3.2.0:copy-resources (default-resources) @ qa-manage-application ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 6 resources
[INFO] Copying 13 resources
[INFO]
[INFO] --- maven-resources-plugin:3.2.0:resources (default-resources) @ qa-manage-application ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 6 resources
[INFO] Copying 13 resources
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-versions) @ qa-manage-application ---
[WARNING] Rule 2: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
The requested JDK version [17,18),[16,17),[15,16),[14,15),[13,14),[12,13),[11,12) is invalid.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.288 s
[INFO] Finished at: 2022-03-03T12:17:38+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce-versions) on project qa-manage-application: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the follow
```

##### **Reproduce the error**

Running `jhipster`, and here are my choices:
```
? Which *type* of application would you like to create? Monolithic application (recommended for simple projects)
? What is the base name of your application? TestApp
? Do you want to make it reactive with Spring WebFlux? No
? What is your default Java package name? com.test.app
? Which *type* of authentication would you like to use? JWT authentication (stateless, with a token)
? Which *type* of database would you like to use? SQL (H2, PostgreSQL, MySQL, MariaDB, Oracle, MSSQL)
? Which *production* database would you like to use? MySQL
? Which *development* database would you like to use? MySQL
? Which cache do you want to use? (Spring cache abstraction) Redis (distributed cache)
? Do you want to use Hibernate 2nd level cache? Yes
? Would you like to use Maven or Gradle for building the backend? Maven
? Do you want to use the JHipster Registry to configure, monitor and scale your application? No
? Which other technologies would you like to use?
? Which *Framework* would you like to use for the client? Vue
? Do you want to generate the admin UI? Yes
? Would you like to use a Bootswatch theme (https://bootswatch.com/)? Default JHipster
? Would you like to enable internationalization support? No
? Please choose the native language of the application Chinese (Simplified)
? Besides JUnit and Jest, which testing frameworks would you like to use? Cypress
? Would you like to install other generators from the JHipster Marketplace? No
```
![image](https://user-images.githubusercontent.com/23058788/156501407-df79c6f8-3d8e-45ca-912f-533690479389.png)

##### **Suggest a Fix**

I'm not sure if it is the reason.

in  `/generators/generator-base-private.js`:

```
  /**
   * Check if Java is installed
   */
  checkJava() {
    if (this.skipChecks || this.skipServer) return;
    const done = this.async();
    exec('java -version', (err, stdout, stderr) => {
      if (err) {
        this.warning('Java is not found on your computer.');
      } else {
        const javaVersion = stderr.match(/(?:java|openjdk) version "(.*)"/)[1];
        if (!javaVersion.match(new RegExp(`(${JAVA_COMPATIBLE_VERSIONS.map(ver => `^${ver}`).join('|')})`))) {
          const [latest, ...others] = JAVA_COMPATIBLE_VERSIONS.reverse();
          this.warning(
            `Java ${others.reverse().join(', ')} or ${latest} are not found on your computer. Your Java version is: ${chalk.yellow(
              javaVersion
            )}`
          );
        }
      }
      done();
    });
  }
```
JAVA_COMPATIBLE_VERSIONS.reverse() changes the origin Array.

##### **JHipster Version(s)**

7.7.0

##### **JHipster configuration**

```
INFO! Switching to JHipster installed locally in current project's node repository (node_modules)

        ██╗ ██╗   ██╗ ████████╗ ███████╗   ██████╗ ████████╗ ████████╗ ███████╗
        ██║ ██║   ██║ ╚══██╔══╝ ██╔═══██╗ ██╔════╝ ╚══██╔══╝ ██╔═════╝ ██╔═══██╗
        ██║ ████████║    ██║    ███████╔╝ ╚█████╗     ██║    ██████╗   ███████╔╝
  ██╗   ██║ ██╔═══██║    ██║    ██╔════╝   ╚═══██╗    ██║    ██╔═══╝   ██╔══██║
  ╚██████╔╝ ██║   ██║ ████████╗ ██║       ██████╔╝    ██║    ████████╗ ██║  ╚██╗
   ╚═════╝  ╚═╝   ╚═╝ ╚═══════╝ ╚═╝       ╚═════╝     ╚═╝    ╚═══════╝ ╚═╝   ╚═╝
                            https://www.jhipster.tech
Welcome to JHipster v7.7.0

Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**


qa-manage-application@0.0.0 D:\Projects\ihealth-qa-solution
`-- generator-jhipster@7.7.0




##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "authenticationType": "jwt",
    "baseName": "QAManageApplication",
    "blueprints": [],
    "buildTool": "maven",
    "cacheProvider": "redis",
    "clientFramework": "vue",
    "clientPackageManager": "npm",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1646211212700,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "devServerPort": 9060,
    "dtoSuffix": "DTO",
    "enableGradleEnterprise": false,
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": false,
    "entitySuffix": "",
    "jhiPrefix": "jhi",
    "jhipsterVersion": "7.7.0",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "languages": ["en", "fr"],
    "messageBroker": false,
    "nativeLanguage": "zh-cn",
    "otherModules": [],
    "packageName": "com.inspur.ihealth.qa.manage",
    "pages": [],
    "prodDatabaseType": "mysql",
    "reactive": false,
    "searchEngine": false,
    "serverPort": "8080",
    "serverSideOptions": [],
    "serviceDiscoveryType": "no",
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "skipUserManagement": false,
    "testFrameworks": ["cypress"],
    "websocket": false,
    "withAdminUi": true
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_291"
Java(TM) SE Runtime Environment (build 1.8.0_291-b10)
Java HotSpot(TM) 64-Bit Server VM (build 25.291-b10, mixed mode)

git version 2.27.0.windows.1

node: v16.14.0

npm: 8.5.2


No change to package.json was detected. No package manager install will be executed.
Congratulations, JHipster execution is complete!
Sponsored with ❤️  by @oktadev.
```

##### **Browsers and Operating System**

Windows 10

- [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/generator-base-private.js
+++ b/generators/generator-base-private.js
@@ -808,8 +808,7 @@ module.exports = class JHipsterBasePrivateGenerator extends Generator {
       } else {
         const javaVersion = stderr.match(/(?:java|openjdk) version "(.*)"/)[1];
         if (!javaVersion.match(new RegExp(`(${JAVA_COMPATIBLE_VERSIONS.map(ver => `^${ver}`).join('|')})`))) {
-          const [latest, ...others] = JAVA_COMPATIBLE_VERSIONS.reverse();
-          JAVA_COMPATIBLE_VERSIONS.reverse();
+          const [latest, ...others] = JAVA_COMPATIBLE_VERSIONS.concat().reverse();
           this.warning(
             `Java ${others.reverse().join(', ')} or ${latest} are not found on your computer. Your Java version is: ${chalk.yellow(
               javaVersion

--- a/generators/generator-base-private.js
+++ b/generators/generator-base-private.js
@@ -809,6 +809,7 @@ module.exports = class JHipsterBasePrivateGenerator extends Generator {
         const javaVersion = stderr.match(/(?:java|openjdk) version "(.*)"/)[1];
         if (!javaVersion.match(new RegExp(`(${JAVA_COMPATIBLE_VERSIONS.map(ver => `^${ver}`).join('|')})`))) {
           const [latest, ...others] = JAVA_COMPATIBLE_VERSIONS.reverse();
+          JAVA_COMPATIBLE_VERSIONS.reverse();
           this.warning(
             `Java ${others.reverse().join(', ')} or ${latest} are not found on your computer. Your Java version is: ${chalk.yellow(
               javaVersion

